Signal/src/view controllers/CountryCodeViewController.m

```diff
@@ -18,10 +18,69 @@ @implementation CountryCodeViewController
 - (void)viewDidLoad {
     [super viewDidLoad];
     [self.navigationController.navigationBar setTranslucent:NO];
-    _countryCodes = [PhoneNumberUtil countryCodesForSearchTerm:nil];
+    _countryCodes = [self countryCodesForSearchTerm:nil];
     self.title    = NSLocalizedString(@"COUNTRYCODE_SELECT_TITLE", @"");
 }
 
+- (NSArray *)countryCodesForSearchTerm:(NSString *)searchTerm {
+    NSMutableArray *result = [NSMutableArray array];
+    for (NSString *countryCode in [PhoneNumberUtil countryCodesForSearchTerm:nil]) {
+        NSString *callingCode = [self callingCodeFromCountryCode:countryCode];
+        if (callingCode != nil &&
+            ![callingCode isEqualToString:@"+0"]) {
+            [result addObject:countryCode];
+        }
+    }
+    return result;
+}
+
+- (NSString *)callingCodeFromCountryCode:(NSString *)countryCode {
+    NSString *callingCode = [PhoneNumberUtil callingCodeFromCountryCode:countryCode];
+    if ([countryCode isEqualToString:@"AQ"]) {
+        // Antarctica
+        callingCode = @"+672";
+    } else if ([countryCode isEqualToString:@"BV"]) {
+        // Bouvet Island
+        callingCode = @"+55";
+    } else if ([countryCode isEqualToString:@"IC"]) {
+        // Canary Islands
+        callingCode = @"+34";
+    } else if ([countryCode isEqualToString:@"EA"]) {
+        // Ceuta & Melilla
+        callingCode = @"+34";
+    } else if ([countryCode isEqualToString:@"CP"]) {
+        // Clipperton Island
+        //
+        // This country code should be filtered.
```

Comment:
Done. Essentially, I can't find reliable info on their calling codes.  I've added the most probable value for the Heard & McDonald Islands, but I'm going to filter Clipperton Island which appears to be uninhabited and not to have a valid calling code.